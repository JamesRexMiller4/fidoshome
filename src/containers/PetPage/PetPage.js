import React from 'react';
import './PetPage.scss';
import Header from '../../components/Header/Header';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const PetPage = ({ animals, id }) =>  {
  const foundPet = animals.find(animal => animal.id === parseInt(id))
    return ( 
      <section>
        <Header />
        <section className='pet-page-section'>
        <img src={foundPet.photos[0].medium} alt='pet-profile' className='pet-profile-pic' />
        <h2 className='pet-name-h2'>{foundPet.name}</h2>
        <section className='details-attributes-section'>
        <ul className='ul'>
          <p className='ul-p'>Attributes:</p>
          <li className='li'>Breed: <span className='span-pet-page'>{foundPet.breeds.primary}</span></li>
          <li className='li'>Color: <span className='span-pet-page'>{foundPet.colors.primary}</span></li>
          <li className='li'>Gender: <span className='span-pet-page'>{foundPet.gender}</span></li>
        </ul>
        <ul className='ul'>
          <p className='ul-p'>Other Details:</p>
          <li className='li'>Spayed/Neutered: <span className='span-pet-page'>{`${foundPet.attributes.spayed_neutered}`}</span></li>
          <li className='li'>House Trained: <span className='span-pet-page'>{`${foundPet.attributes.house_trained}`}</span></li>
          <li className='li'>Up-to-Date on Shots: <span className='span-pet-page'>{`${foundPet.attributes.shots_current}`}</span></li>
        </ul>
        <ul className='ul'>
          <p className='ul-p'>Contact:</p>
          <li className='li'>Email: <span className='span-pet-page'>{foundPet.contact.email}</span></li>
          <li className='li'>Phone: <span className='span-pet-page'>{foundPet.contact.phone}</span></li>
          <li className='li'>Address: <span className='span-pet-page'>{`${foundPet.contact.address.address1} ${foundPet.contact.address.city}, ${foundPet.contact.address.state} ${foundPet.contact.address.postcode}`}</span></li>
        </ul>
        </section>
        </section>
      </section>
    );
}

export const mapStateToProps = state => ({
  animals: state.animals
})

export default connect(mapStateToProps)(PetPage);

PetPage.propTypes = {
  animals: PropTypes.array,
  id: PropTypes.string
}