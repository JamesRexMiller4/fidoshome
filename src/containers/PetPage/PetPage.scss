.pet-profile-pic {
  border: 10px solid;
  border-color: rgb(92, 10, 186);
  border-radius: 5px;
  margin-top: 10px;
  height: 40vh;
}

.pet-page-section {
  background-color: white;
  // height: 80vh;
}

.details-attributes-section {
  display: flex;
  justify-content: space-around ;
  margin-top: 30px;
}

.pet-name-h2 {
  font-family: 'Libre Baskerville', serif;
  font-weight: bold;
  font-size: 3rem;
  margin-top: 20px;
}

.ul {
  list-style: none;
}

.ul-p {
  font-family: 'Libre Baskerville', serif;
  font-weight: bold;
  font-size: 1.5rem
}

.li {
  margin-top: 10px;
  font-weight: bold;
  text-align: justify;
}
.span-pet-page {
  font-weight: normal;
  font-size: 1.25rem;
  color: rgb(73, 73, 73);
}