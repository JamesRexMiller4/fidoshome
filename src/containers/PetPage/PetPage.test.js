import React from 'react';
import {shallow} from 'enzyme';
import { PetPage, mapStateToProps } from './PetPage';
import VeryGoodBoy from '../../images/verygoodboy.PNG';

describe('PetPage', () => {
let wrapper, mockData;

  beforeEach(() => {
    mockData = [{
      photos: [{medium: VeryGoodBoy}],
      id: 1, 
      name: 'Keiko',
      breeds: {primary: 'Akita'},
      colors: {primary: 'White, Black, Brown'},
      gender: 'female',
      attributes: {spayed_neutered: true, house_trained: true, shots_current: true},
      age: 'Older than time itself',
      contact: {address: {address1: 'All Dogs Go to Heaven Ct.', city: 'Big Sky', state: 'IL', postcode: '60614'}, email: 'jarvisblargus@gmail.com', phone: '773-312-9456'}
    },
    {
      photos: [{medium: VeryGoodBoy}],
      id: 2, 
      name: 'Buster',
      breeds: {primary: 'Dachshund'},
      colors: {primary: 'Brown, Dappled'},
      gender: 'male',
      attributes: {spayed_neutered: false, house_trained: true, shots_current: true},
      age: 'Older than time itself',
      contact: {address: {address1: 'All Dogs Go to Heaven Ct.', city: 'Big Sky', state: 'IL', postcode: '60614'}, email: 'jarvisblargus@gmail.com', phone: '773-312-9456'}
    }];

    wrapper = shallow(<PetPage 
      animals={mockData}
      id={1}
    />)
  });

  it('should match the snapshot with all the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

    describe('mapStateToProps', () => {
      it('should return an object with animals data', () => {
        const mockState = {
          animals: mockData
        };
        const expected = {animals: mockData};
        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expected);
      });
    });
});