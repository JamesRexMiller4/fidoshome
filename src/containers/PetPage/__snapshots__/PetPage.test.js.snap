// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetPage should match the snapshot with all the data passed through 1`] = `
<section>
  <Header />
  <section
    className="pet-page-section"
  >
    <img
      alt="pet-profile"
      className="pet-profile-pic"
      src="verygoodboy.PNG"
    />
    <h2
      className="pet-name-h2"
    >
      Keiko
    </h2>
    <section
      className="details-attributes-section"
    >
      <ul
        className="ul"
      >
        <p
          className="ul-p"
        >
          Attributes:
        </p>
        <li
          className="li"
        >
          Breed: 
          <span
            className="span-pet-page"
          >
            Akita
          </span>
        </li>
        <li
          className="li"
        >
          Color: 
          <span
            className="span-pet-page"
          >
            White, Black, Brown
          </span>
        </li>
        <li
          className="li"
        >
          Gender: 
          <span
            className="span-pet-page"
          >
            female
          </span>
        </li>
      </ul>
      <ul
        className="ul"
      >
        <p
          className="ul-p"
        >
          Other Details:
        </p>
        <li
          className="li"
        >
          Spayed/Neutered: 
          <span
            className="span-pet-page"
          >
            true
          </span>
        </li>
        <li
          className="li"
        >
          House Trained: 
          <span
            className="span-pet-page"
          >
            true
          </span>
        </li>
        <li
          className="li"
        >
          Up-to-Date on Shots: 
          <span
            className="span-pet-page"
          >
            true
          </span>
        </li>
      </ul>
      <ul
        className="ul"
      >
        <p
          className="ul-p"
        >
          Contact:
        </p>
        <li
          className="li"
        >
          Email: 
          <span
            className="span-pet-page"
          >
            jarvisblargus@gmail.com
          </span>
        </li>
        <li
          className="li"
        >
          Phone: 
          <span
            className="span-pet-page"
          >
            773-312-9456
          </span>
        </li>
        <li
          className="li"
        >
          Address: 
          <span
            className="span-pet-page"
          >
            All Dogs Go to Heaven Ct. Big Sky, IL 60614
          </span>
        </li>
      </ul>
    </section>
  </section>
</section>
`;
